internal/gensupport/error.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2020 Google LLC. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gensupport
+
+import (
+	"errors"
+
+	"github.com/googleapis/gax-go/v2/apierror"
+	"google.golang.org/api/googleapi"
+)
+
+// WrapError creates an apierror.APIError from err (one that
+// does not wrap err), wraps it in err, and returns err. If
+// err is not a googleapi.Error (or a gRPC Status), it returns
+// err without modification.
+func WrapError(err error) error {
+	if apiError, ok := apierror.ParseError(err, false); ok {
```

Comment:
nit: reverse the logic and shift right the !ok case

Reply:
Refactored to remove a line of code and avoid early return. Is it too confusing now?

Reply:
I think this is fine, but I might scoot line 20 up to top of func if I am being nitpicky. I like to see if checks directly proceeding the line the are referencing. ok is declared at 19 and checked at 21.